From f99b932cf2175c1c59f42a95358145e4c78580ea Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinh.nguyen@intel.com>
Date: Wed, 6 Nov 2019 10:01:03 -0600
Subject: [PATCH 051/151] HSD #14010378260-1: Add nand_x_clk and nand_ecc_clk

commit  5326e801cfa7ae2c4c922d683f853861e4ace0cb from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

Add the nand_x_clk and nand_ecc_clk. Also the nand_x_clk's parent is
the l4_mp_clk, not the l4_main_clk.

Signed-off-by: Dinh Nguyen <dinh.nguyen@intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 drivers/clk/socfpga/clk-agilex.c         | 6 +++++-
 include/dt-bindings/clock/agilex-clock.h | 4 +++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/socfpga/clk-agilex.c b/drivers/clk/socfpga/clk-agilex.c
index ba5ba8866d1b..7f33dbc308ff 100644
--- a/drivers/clk/socfpga/clk-agilex.c
+++ b/drivers/clk/socfpga/clk-agilex.c
@@ -148,8 +148,12 @@ static const struct stratix10_gate_clock agilex_gate_clks[] = {
 	  8, 0, 0, 0, 0, 0, 0},
 	{ AGILEX_SPI_M_CLK, "spi_m_clk", "l4_mp_clk", NULL, 1, 0, 0x7C,
 	  9, 0, 0, 0, 0, 0, 0},
-	{ AGILEX_NAND_CLK, "nand_clk", "l4_main_clk", NULL, 1, 0, 0x7C,
+	{ AGILEX_NAND_X_CLK, "nand_x_clk", "l4_mp_clk", NULL, 1, 0, 0x7C,
 	  10, 0, 0, 0, 0, 0, 0},
+	{ AGILEX_NAND_CLK, "nand_clk", "nand_x_clk", NULL, 1, 0, 0x7C,
+	  10, 0, 0, 0, 0, 0, 4},
+	{ AGILEX_NAND_ECC_CLK, "nand_ecc_clk", "nand_x_clk", NULL, 1, 0, 0x7C,
+	  10, 0, 0, 0, 0, 0, 4},
 };
 
 static int agilex_clk_register_c_perip(const struct stratix10_perip_c_clock *clks,
diff --git a/include/dt-bindings/clock/agilex-clock.h b/include/dt-bindings/clock/agilex-clock.h
index 4c2a5e80ea86..f751aad4dafc 100644
--- a/include/dt-bindings/clock/agilex-clock.h
+++ b/include/dt-bindings/clock/agilex-clock.h
@@ -64,6 +64,8 @@
 #define AGILEX_SDMMC_CLK		48
 #define AGILEX_SPI_M_CLK		49
 #define AGILEX_USB_CLK			50
-#define AGILEX_NUM_CLKS			51
+#define AGILEX_NAND_X_CLK		51
+#define AGILEX_NAND_ECC_CLK		52
+#define AGILEX_NUM_CLKS			53
 
 #endif	/* __AGILEX_CLOCK_H */
-- 
2.26.1

